src/themes/ec/mixins/_typography.scss

```diff
@@ -0,0 +1,70 @@
+@use 'sass:map';
+@use 'sass:string';
+@use '@ecl/grid/mixins/breakpoints';
+
+$font-size-map: (
+  '8xl': '7xl',
+  '6xl': '5xl',
+  '3xl': '2xl',
+  '2xl': '2xl',
+  'xl': 'xl',
+  'l': 'l',
+  'm': 'm',
+  's': 's',
+  'xs': 'xs',
+);
+
+// Helper to check if value is a CSS variable
+@function is-var($value) {
+  @return string.index(#{$value}, 'var(--') == 1;
+}
+
+@function extract-token($value) {
+  @if is-var($value) {
+    $start: string.index(#{$value}, '--f-') + 4;
+    $end: string.index(#{$value}, ')');
+
+    @return string.slice(#{$value}, $start, $end - 1);
+  }
+
+  @return $value;
+}
+
+/// Responsive font
+/// @param {String} $desktop-size - Font size for desktop (e.g. '3xl' or var(--f-3xl))
+/// @param {String | null} $mobile-size - Optional override for mobile font size
+/// @param {String} $breakpoint - Breakpoint for desktop font size (default: 'm')
+/// @param {Number | String | null} $font-weight - Optional font weight (e.g. 300, 500, bold)
+@mixin responsive-font(
+  $desktop-size,
+  $mobile-size: null,
+  $breakpoint: 'm',
+  $font-weight: 300
+) {
+  $desktop-key: extract-token($desktop-size);
+  $mobile-key: if(
+    $mobile-size,
+    extract-token($mobile-size),
+    map.get($font-size-map, $desktop-key)
+  );
+
+  @if not $mobile-key {
+    $mobile-key: $desktop-key;
+  }
+
+  $resolved-mobile: var(--f-#{$mobile-key});
+  $resolved-desktop: var(--f-#{$desktop-key});
+
+  font: $resolved-mobile;
+
+  @if $font-weight {
+    font-weight: $font-weight;
+  }
+
+  // Only add desktop font-size at breakpoint if it differs from mobile
+  @if $resolved-desktop != $resolved-mobile {
+    @include breakpoints.up($breakpoint) {
+      font: $resolved-desktop;
```

Comment:
we should have the font weight for mobile too

Reply:
added font-weight

Reply:
also added letter spacing too, if there is another map for letter spacing, that can be overwritter in the param